### ISSUE TITLE
Preview thumbnail of attached image file is not shown in UI

### ISSUE THREAD

**Issue by leplatrem** [Kinto/kinto#3386]
<img width="1425" alt="Screenshot 2024-02-07 at 14 04 16" src="https://github.com/Kinto/kinto/assets/546692/09b0caf0-4deb-4150-a26a-dc5cb2b3ac5b">

The URL is correct but it is blocked by `Content-Security-Policy: The page’s settings blocked the loading of a resource `

See https://github.com/Kinto/kinto/blob/e8e55d19616d67ee1c9f7d7baa6a67986fcaf058/kinto/plugins/admin/views.py#L30-L40

(could have opened this issue but used remote-settiungs

**Comment by Angie-newbie** [comment]
Hello! 
I would pick up this issue and work on it :)

**Comment by leplatrem** [comment]
Awesome!

**Comment by Angie-newbie** [comment]
Hi @leplatrem ,

I am new to this system, can you please provide the list of steps to replicate the issue? :)

**Comment by leplatrem** [comment]
Of course, I'm sorry, I'll try to give as much info as possible.

First make sure you can run the project locally (check out the docs) but in brief it would be:

```
git clone git@github.com:fork/kinto.git
cd kinto/
make pull-kinto-admin
make config/kinto.ini
```
You can pick the memory backend to skip postgresql stuff. But you'll have to repeat the account creation on each restart.

Now, we'll need the attachment plugin, it's installed by default in the docker container, but not when developping from sources:

```
.venv/bin/pip install kinto-attachment
```
Enable it in the `config/kinto.ini`:
```diff
 # https://kinto.readthedocs.io/en/latest/configuration/settings.html
@@ -35,6 +35,7 @@
 kinto.includes = kinto.plugins.default_bucket
                  kinto.plugins.admin
                  kinto.plugins.accounts
+                 kinto_attachment
 #                kinto.plugins.history
 #                kinto.plugins.quotas
+
+kinto.attachment.base_url = http://localhost:9999/
+kinto.attachment.base_path = /tmp

```

In one terminal, you run a server to deliver the attachment files:
```
cd /tmp
python -m http.server 9999
```

In another terminal, you serve the Kinto application:
```
make serve
```

Create the admin account:

```
curl -X POST http://127.0.0.1:8888/v1/accounts \
     -H "Content-Type: application/json" \
     -d '{"data": {"id": "admin", "password": "s3cr3t"}}'
```

Create a default collection to reproduce the bug:

```
curl -X PUT 'http://127.0.0.1:8888/v1/buckets/default/collections/bug-3386' \
     --user admin:s3cr3t \
     -H 'Content-Type: application/json' \
     --data-raw '{
       "data": {
         "schema": {
           "type": "object",
           "properties": {
             "title": { "type": "string" }
           }
         },
         "displayFields": ["title"],
         "attachment": {
           "enabled": true,
           "required": true
         },
         "id": "bug-3386"
       }
     }'

```

(I give the commands to make it easier to replay)

Now:
- Navigate to http://127.0.0.1:8888/v1/admin/
- Login with your credentials
- Create a record on the collection, add an image attachment
- Edit the record again
- Observe the message in the console: `Content-Security-Policy: The page’s settings blocked the loading of a resource (img-src) at http://localhost:9999/84736286-3dfc-4dfc-95d8-ba1db1e0579d.jpg because it violates the following directive: “img-src data: 'self'”`

`self` here is too restrictive, the admin UI should be allowed to serve images from the `kinto.attachment.base_url` URL too.

See https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP

Hope you make it through all of this! Have fun investigating and hacking!

**Comment by Angie-newbie** [comment]
![Image](https://github.com/user-attachments/assets/4c8392e7-f5dd-4d23-8b97-aa306a9af55a)


Hi @leplatrem , Thank you for the setup steps! 

After I attached a image on the record, and click "edit the record", the screenshot above is what I saw.
Which is really different to yours, and seems like the image file is gone.

Just wondering why the view is so different with the one you attached for this issue?

**Comment by leplatrem** [comment]
Can you check whether the attachments are enabled on the server?
Navigate to `Home` and check whether `attachments` is shown in the server information

(I just opened this issue to limit confusion https://github.com/Kinto/kinto-admin/issues/3426)

**Comment by leplatrem** [comment]
@Angie-newbie are you still interested to work on this issue?

**Comment by Angie-newbie** [comment]
Hi @leplatrem !
Yes, but the problem that I had got is that we cannot replicate this issue as the another issue need to be solved first.

**Comment by leplatrem** [comment]
> Can you check whether the attachments are enabled on the server?

🆙

**Comment by Angie-newbie** [comment]
The attachments are not enabled on the server....
I had tried couple different images, but they all not shown in the server information

**Comment by leplatrem** [comment]
First you need to enable the attachment plugin on the server.

In your `kinto.ini` configuration, make sure you have: 

```ini
kinto.includes = ...
                 kinto_attachment

kinto.attachment.base_path = /tmp
```

The attachment section should show up in the the server information home page on the admin, and the file upload should up in the forms

**Comment by Angie-newbie** [comment]
Hi @leplatrem ,
Yes! I did have those plugin in the file, but it still doesn't work

![Image](https://github.com/user-attachments/assets/00ec94f1-2f97-4dc1-af35-2dd5fe1802a4)

**Comment by leplatrem** [comment]
Weird, I just tried from scrach these commands, and can reproduce:


```
cd /tmp
git clone git@github.com:Kinto/kinto.git
cd kinto
git checkout 20.1.0
make install-dev
make pull-kinto-admin
.venv/bin/pip install kinto-attachment
.venv/bin/kinto init --ini test.ini
```
